src/pages/loading-waiting-states/contextual-spinner/ContextualSpinner.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { makeStyles, useTheme, Theme } from '@material-ui/core/styles';
+import { AppBar, Toolbar, Hidden, IconButton, Typography, Button, Fab, CircularProgress } from '@material-ui/core';
+import { TOGGLE_DRAWER } from '../../../redux/actions';
+import { useDispatch } from 'react-redux';
+import { Menu, PlayArrow } from '@material-ui/icons';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+        backgroundColor: theme.palette.background.paper,
+        minHeight: '100vh',
+    },
+    appbarRoot: {
+        padding: 0,
+    },
+    toolbarGutters: {
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    exampleContainer: {
+        margin: `${theme.spacing(3)}px ${theme.spacing(2)}px`,
+    },
+    description: {
+        marginBottom: theme.spacing(2),
+    },
+    loginButton: {
+        marginBottom: theme.spacing(2),
+        width: 90,
+        height: 36,
+    },
+    startRoutineButton: {
+        width: 156,
+        height: 56,
+        borderRadius: 29,
+        transition: 'width 300ms',
```

Comment:
you can use `theme.transitions.create` to set this up without hardcoding a duration (https://next.material-ui.com/customization/transitions/)